tests/test_dummy.py

```diff
@@ -0,0 +1,309 @@
+    dg = identify_dummy_groups(mol, core)
+    assert_set_equality(dg, [{3, 4}])
+
+    mol = Chem.MolFromSmiles("C1CC2OOC12")
+    core = [0, 1, 2, 5]
+    dg = identify_dummy_groups(mol, core)
+    assert_set_equality(dg, [{3, 4}])
+
+
+def assert_anchor_group_equality(a_groups, b_groups):
+    frozen_a = [tuple(a) for a in a_groups]
+    frozen_b = [tuple(b) for b in b_groups]
+    assert frozenset(frozen_a) == frozenset(frozen_b)
+
+
+def test_identify_anchor_groups():
+
+    mol = Chem.MolFromSmiles("C1CC1F")
+    core = [0, 1, 2]
+    groups_of_1, groups_of_2, groups_of_3 = identify_anchor_groups(mol, core, 0)
+    assert_anchor_group_equality(groups_of_1, [[0]])
+    assert_anchor_group_equality(groups_of_2, [[0, 1], [0, 2]])
+    assert_anchor_group_equality(groups_of_3, [[0, 1, 2], [0, 2, 1]])
+
+    groups_of_1, groups_of_2, groups_of_3 = identify_anchor_groups(mol, core, 1)
+    assert_anchor_group_equality(groups_of_1, [[1]])
+    assert_anchor_group_equality(groups_of_2, [[1, 0], [1, 2]])
+    assert_anchor_group_equality(groups_of_3, [[1, 0, 2], [1, 2, 0]])
+
+    groups_of_1, groups_of_2, groups_of_3 = identify_anchor_groups(mol, core, 2)
+    assert_anchor_group_equality(groups_of_1, [[2]])
+    assert_anchor_group_equality(groups_of_2, [[2, 1], [2, 0]])
+    assert_anchor_group_equality(groups_of_3, [[2, 1, 0], [2, 0, 1]])
+
+    mol = Chem.MolFromSmiles("NCCF")
+    core = [1, 2]
+    groups_of_1, groups_of_2, groups_of_3 = identify_anchor_groups(mol, core, 1)
+    assert_anchor_group_equality(groups_of_1, [[1]])
+    assert_anchor_group_equality(groups_of_2, [[1, 2]])
+    assert_anchor_group_equality(groups_of_3, [])
+
+    mol = Chem.MolFromSmiles("C(C)(C)(C)C")
+    core = [0]
+    groups_of_1, groups_of_2, groups_of_3 = identify_anchor_groups(mol, core, 0)
+    assert_anchor_group_equality(groups_of_1, [[0]])
+    assert_anchor_group_equality(groups_of_2, [])
+    assert_anchor_group_equality(groups_of_3, [])
+
+    mol = Chem.MolFromSmiles("C(C)(F)CC")
+    core = [1, 0, 4, 3]
+    groups_of_1, groups_of_2, groups_of_3 = identify_anchor_groups(mol, core, 0)
+    assert_anchor_group_equality(groups_of_1, [[0]])
+    assert_anchor_group_equality(groups_of_2, [[0, 1], [0, 3]])
+    assert_anchor_group_equality(groups_of_3, [[0, 3, 4]])
+
+
+def test_enumerate_anchor_groups():
+    mol = Chem.MolFromSmiles("C1CC2OOC12")
+    core = [0, 1, 2, 5]
+    groups_of_1, groups_of_2, groups_of_3 = enumerate_anchor_groups(mol, core, [3, 4])
+    assert_anchor_group_equality(groups_of_1, [[2], [5]])
+    assert_anchor_group_equality(groups_of_2, [[2, 1], [2, 5], [5, 0], [5, 2]])
+    assert_anchor_group_equality(groups_of_3, [[2, 1, 0], [2, 5, 0], [5, 0, 1], [5, 2, 1]])
+
+    # aspirin
+    mol = Chem.MolFromSmiles("CC(=O)OC1=CC=CC=C1C(=O)O")
+    core = [3, 4, 5, 6, 7, 8, 9, 10]
+    groups_of_1, groups_of_2, groups_of_3 = enumerate_anchor_groups(mol, core, [0, 1, 2, 11, 12])
+    assert_anchor_group_equality(groups_of_1, [[3], [10]])
```

Comment:
Should this validate that groups_of_2 and groups_of_3 are empty sets?

Reply:
They are not empty sets. I'll add the members more explicitly